src/external/nonzero_support.rs

```diff
@@ -0,0 +1,121 @@
+//! A wrapper type for non-zero UUIDs that provides a more memory-efficient
+//! `Option<NonZeroUuid>` representation.
+
+use std::convert::TryFrom;
+use std::fmt;
+use std::num::NonZeroU128;
+use std::ops::Deref;
+
+use crate::Uuid;
+
+/// A UUID that is guaranteed not to be the nil UUID.
+///
+/// This is useful for representing optional UUIDs more efficiently, as `Option<NonZeroUuid>`
+/// takes up the same space as `Uuid`.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct NonZeroUuid(NonZeroU128);
+
+/// Error returned when attempting to create a `NonZeroUuid` from a nil UUID.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct NonZeroUuidError;
```

Comment:
This should be a struct instead of a newtype, so that dependents of `uuid` can't construct it

Reply:
Remove this for now, since used only on TryFrom trait